pytorch_lightning/callbacks/early_stopping.py

```diff
@@ -53,6 +53,8 @@ class EarlyStopping(Callback):
             monitored has stopped decreasing and in ``'max'`` mode it will stop when the quantity
             monitored has stopped increasing.
         strict: whether to crash the training if `monitor` is not found in the validation metrics.
+        check_on_train_epoch_end: whether to run early stopping at the end of the training epoch.
```

Comment:
name suggestion: In ModelCheckpoint, we have arguments called `every_n_train_steps` and `every_n_val_epochs`. For consistency, I propose `every_train_epoch`.

Reply:
The checkpoint callback also has the frequency as part of the name. I thought the `on_train_epoch_end` would be clear to map to which callback hooks runs, but open to naming!

Reply:
I personally prefer Ananth's option